python/taichi/lang/impl.py

```diff
@@ -136,6 +139,9 @@ def subscript(value, *_indices, skip_reordered=False, get_ref=False):
     if isinstance(value, np.ndarray):
         return value.__getitem__(_indices)
 
+    if isinstance(value, Expr):
+        return make_index_expr(value.ptr, _indices)
```

Comment:
This part seems already covered by https://github.com/taichi-dev/taichi/blob/8c8b74db776ab89deea0aa0dc7bc2c3c27adea10/python/taichi/lang/impl.py#L233-L242